src/hkdf.rs

```diff
@@ -39,19 +39,19 @@ use hkdf::hmac::Hmac;
 ///
 /// # fn main() {
 ///     let ikm = b"02345678001234567890123456789012";
-///     let domain = b"my_app";
+///     let info = b"my_app";
 ///     let salt = b"some_salt";
 
-///     let my_keypair = hkdf_generate_from_ikm::<Sha3_256, BLS12381KeyPair>(ikm, salt, Some(domain)).unwrap();
-///     let native_sk = blst::min_sig::SecretKey::key_gen_v4_5(ikm, salt, domain).unwrap();
+///     let my_keypair = hkdf_generate_from_ikm::<Sha3_256, BLS12381KeyPair>(ikm, salt, info).unwrap();
+///     let native_sk = blst::min_sig::SecretKey::key_gen_v4_5(ikm, salt, info).unwrap();
 
 ///     assert_ne!(native_sk.to_bytes(), my_keypair.private().as_bytes());
 /// # }
 /// ```
-pub fn hkdf_generate_from_ikm<'a, H: OutputSizeUser, K: KeyPair>(
-    ikm: &[u8],             // IKM (32 bytes)
-    salt: &[u8],            // Optional salt
-    info: Option<&'a [u8]>, // Optional domain
+pub fn hkdf_generate_from_ikm<H, K>(
+    ikm: &[u8],  // IKM (32 bytes).
+    salt: &[u8], // Salt (can be empty).
+    info: &[u8], // Info (can be empty).
```

Comment:
Indeed for the `info` field both the underlying dependency lib and the rfc, do not optionally expect it and can be empty. This is different from the salt which defaults to a HashLen zero byte string.

Per RFC:

       salt     optional salt value (a non-secret random value);
               **if not provided, it is set to a string of HashLen zeros.**
               
       info     optional context and application specific information
               **(can be a zero-length string)**